### PR TITLE
Add: Load API Definition from the configmap

### PR DESCRIPTION
**Pull request by BLasan**
## Purpose
Fetch API Definition of an API using a specific path

## Approach
- Read the definition from the corresponding configmap
- Load the definition to the enforcer

### CODE REVIEW

File: adapter/internal/oasparser/envoyconf/routes_with_clusters.go [hunk 1]
```diff
@@ -114,6 +114,26 @@ func CreateRoutesWithClusters(adapterInternalAPI model.AdapterInternalAPI, inter
...
+		URLType: "https",
+	}
+	endpointForAPIDefinitions = append(endpointForAPIDefinitions, *endpoint)
+	endpointCluster := model.EndpointCluster{
+		Endpoints: endpointForAPIDefinitions,
+	}
+
+	cluster, address, err := processEndpoints("api_definition_cluster", &endpointCluster, timeout, "")
+	if err != nil {
+		logger.LoggerOasparser.ErrorC(logging.GetErrorByCode(2239, apiTitle, apiVersion, apiDefinitionPath, err.Error()))
```
Comment (AmaliMatharaarachchi) [inline adapter/internal/oasparser/envoyconf/routes_with_clusters.go:132]: I couldn't find respective error for code `2239`

**Review by tharindu1st — DISMISSED** [review]

File: adapter/internal/oasparser/envoyconf/routes_with_clusters.go (hunk 1 above)
Comment (BLasan) [reply adapter/internal/oasparser/envoyconf/routes_with_clusters.go:132]: https://github.com/wso2/apk/blob/main/adapter/internal/logging/logging_constant.go#L431

File: test/postman-tests/resources/collections/Test_Runtime_API_Operations-collection.json [hunk 2]
```diff
@@ -447,6 +447,71 @@
...
+				"method": "GET",
+				"header": [
+					{
+						"key": "Host",
+						"value": "api.am.wso2.com",
+						"type": "default"
+					}
+				],
+				"url": {
+					"raw": "{{gatewayBaseURl}}/api/runtime/1.0.0/swagger.json",
```
Comment (CrowleyRajapakse) [inline test/postman-tests/resources/collections/Test_Runtime_API_Operations-collection.json:501]: is this path correct? this test also seems to fail with 503 service not available

Comment (BLasan) [reply test/postman-tests/resources/collections/Test_Runtime_API_Operations-collection.json]: This test is removed. Will add this in a separate PR. Also, these are the failing tests.
``` TestApiRateLimiting:1 | expected response to have status code 200 but got 503
TestRequestInterceptor:1 | expected undefined to equal 'Interceptor-header-value'
TestRequestInterceptor:1 | expected undefined to equal 'Interceptor-header-value'
TestRequestInterceptor:1 | expected undefined to equal 'Interceptor-Response-header-value'
TestRequestInterceptor:1 | expected undefined to equal 'Interceptor-Response-header-value'
TestRequestInterceptor:1 | expected undefined to equal 'Interceptor-header-value'
TestRequestInterceptor:1 | expected undefined to equal 'Interceptor-header-value'
TestRequestInterceptor:1 | expected response to have status code 200 but got 429```

I ran rate limit tests locally and all were passed without any error. Regarding `Interceptor-header-value` issue, this occurs in this PR as well. https://github.com/wso2/apk/actions/runs/5199491115/jobs/9377265813?pr=1227

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Pull request merged**